src/machinelist.cpp

```diff
@@ -1576,37 +1568,40 @@ void MachineList::parse()
 							hiddenItemHash.insert(machineItem, true);
 						// find year & manufacturer and determine ROM/CHD requirements
 						bool endMachine = false;
-						int i = lineCounter;
+
 						QString machineYear(trQuestionMark), machineManufacturer(trQuestionMark), machinePlayers(trQuestionMark), machineDrvStat(trQuestionMark);
 						bool yearFound = false, manufacturerFound = false, hasROMs = false, hasCHDs = false, playersFound = false, statusFound = false;
-						QString endMark("</machine>");
-						while ( !endMachine ) {
-							QString xmlLine(xmlLines.at(i));
-							if ( xmlLine.contains("<year>") ) {
-								machineYear = xmlLine.simplified().remove("<year>").remove("</year>");
+						while ( xmlMachineEntry.readNextStartElement() && !endMachine ) {
+							if ( xmlMachineEntry.name() == "description" ) {
+								descriptionElement = xmlMachineEntry.readElementText();
+								machineDescription = descriptionElement.replace("&amp;", "&").replace("&lt;", "<").replace("&gt;", ">").replace("&quot;", "\"").replace("&apos;", "'");
+							} else if ( xmlMachineEntry.name() == "year" ) {
+								machineYear = xmlMachineEntry.readElementText();
 								yearFound = true;
-							} else if ( xmlLine.contains("<manufacturer>") ) {
-								machineManufacturer = xmlLine.simplified().remove("<manufacturer>").remove("</manufacturer>").replace("&amp;", "&").replace("&lt;", "<").replace("&gt;", ">").replace("&quot;", "\"").replace("&apos;", "'");
+							} else if ( xmlMachineEntry.name() == "manufacturer" ) {
+								machineManufacturer = xmlMachineEntry.readElementText().replace("&amp;", "&").replace("&lt;", "<").replace("&gt;", ">").replace("&quot;", "\"").replace("&apos;", "'");
```

Comment:
More XML character entity decoding you shouldn’t be doing here.

Reply:
Done